examples/starboard_clone/src/memhandl.c

```diff
@@ -0,0 +1,362 @@
+  sqlite3_prepare_v2(sbclone_DB, sql, strlen(sql), &stmt, NULL);
+  char message_id[20];
+  char* tmp = message_id;
+  snowflake_to_char(tmp, event->message_id);
+  sqlite3_bind_text(stmt, 1, message_id, strlen(message_id), SQLITE_STATIC);
+  int rc = sqlite3_step(stmt);
+
+  // This is set, if the submission was checked to be already sent to the board channel
+  int isposted = 0;
+
+  // Fetch the message
+  struct discord_message message;
+  struct discord_ret_message message_ret = { .sync = &message };
+  CCORDcode code;
+
+  // Get the message object and make check.
+  code = discord_get_channel_message(client, event->channel_id, event->message_id, &message_ret);
+  assert(CCORD_OK == code && "Couldn't fetch message");
+
+  struct discord_user client_user;
+  struct discord_ret_user client_user_ret = { .sync = &client_user };
+
+  discord_get_current_user(client, &client_user_ret);
+  if(message.author->id == client_user.id) return;
+
+  char user_id[83]; // Don't even try to ask me, how 83 and up, are the only sizes that make it not crash
+                // Don't even try to say to me that "it works with like 20", it does only in debug mode
+                // I have NO idea why???? or how?????, you gotta trust me on that
+
+  switch(rc) {
+    case SQLITE_ROW:
+      isposted = sqlite3_column_int(stmt, 4);
+      break;
+    case SQLITE_DONE:
+      snowflake_to_char(user_id, message.author->id);
+
+      sbclone_create_row(message_id, user_id);
+      break;
+    default:
+      // Error
+      fprintf(stderr, "Couldn't query post.\n");
+      break;
+  }
+  sqlite3_finalize(stmt);
+
+  // Checks if its already posted
+  if(isposted == 1) return;
+  // Check if it should add a star reaction
+  /*if(message.reactions->size == 0) {
+    // Add the star reaction
+    discord_create_reaction(client, message.channel_id, message.id, 0, "⭐", NULL);
+  }*/
+  int i;
+  while(i < message.reactions->size) {
+    // Check if the emoji is a star
+    if(strcmp(message.reactions->array[i].emoji->name, "⭐") == 0) {
+      // Check if the message surpassed the minimum reaction count to be sent to the board
+      if(message.reactions->array[0].count < 10) return; //! You can change me to whatever you want <3
+      break;
+    }
+    i++;
+  }
+  // Check if the message surpassed the minimum reaction count to be sent to the board
+  //if(message.reactions->array[0].count < 1) return; //! You can change me to whatever you want <3
+
+  // Post it.
+  sbclone_post_to_channel(client, &message, message_id, &event->message_id, &event->channel_id, &event->guild_id);
+}
+
+int sbclone_check_if_star_channel(u64snowflake channel_id) {
+  sqlite3_stmt* stmt;
+  char* sql = "SELECT * FROM channels WHERE channelid = ?";
+  sqlite3_prepare_v2(sbclone_DB, sql, strlen(sql), &stmt, NULL);
+  char char_channel_id[20];
+  snowflake_to_char((char *)&char_channel_id, channel_id);
+  sqlite3_bind_text(stmt, 1, char_channel_id, strlen(char_channel_id), SQLITE_STATIC);
```

Comment:
Nit: `snprintf` returns the length of the string, you can use it to avoid re-calculating it